pandas/core/apply.py

```diff
@@ -1337,7 +1337,7 @@ def _make_unique_kwarg_list(
 
 
 def relabel_result(
-    result: FrameOrSeries,
+    result: DataFrame | Series,
```

Comment:
why are we not able to use the alias, this is the entire point of it (i mean if we want to switch exclusively to `DataFrame | Series` i guess its ok, but we have an alias on purpose.

Reply:
`FrameOrSeries = TypeVar("FrameOrSeries", bound="NDFrame")` 
A `TypeVar` needs to be bound which isn't the case here.

Reply:
Might be helpful to rename `FrameOrSeries` to `NDFrameT`

Reply:
ahh ok, then lets do that

Reply:
in this PR or a followup?

I don't mind adding more (I'm not the one reviewing it)

Reply:
follow-up ok. we have the naming for historical reasons and IIRC other reviewers have strong views on the naming, so IMO best not to mix in anything that could be controversial.